### ISSUE TITLE
Monad[Free[Id, ?]] cannot be found

### ISSUE THREAD

**Issue by johnynek** [typelevel/cats#2281]
```
[error] /Users/oscar/oss/cats/free/src/test/scala/cats/free/FreeSuite.scala:18:13: could not find implicit value for parameter e: cats.Monad[[β$1$]cats.free.Free[[A]A,β$1$]]
[error]   implicitly[Monad[Free[Id, ?]]]                                     
[error]             ^                                                                                 
[error] one error found                                                                                 
[error] (freeJVM / Test / compileIncremental) Compilation failed 
```

Note that there is no problem with `implicitly[Monad[Free[Option, ?]]]`. I wonder if this could be related to the possible compiler bug found in #2276

**Comment by johnynek** [comment]
note I discovered this in #2279 where I found that I also could not resolve `Defer[Free[Id, ?]]`

**Comment by ceedubs** [comment]
Scalac generally has issues with inference with `Id`. I think that this is a bit different than #2276. I think that this just requires specialized instances for `Id` like we've done [here](https://github.com/typelevel/cats/blob/master/core/src/main/scala/cats/data/Kleisli.scala#L177) for `Kleisli`.

**Comment by kailuowang** [comment]
fixed by @barambani in #2383
